src/entity/queue.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{
+    json_newtype,
+    queue::{Job, JobError},
+};
+use sea_orm::{entity::prelude::*, Set};
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+use time::OffsetDateTime;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum JobResult {
+    Complete,
+    Child(Uuid),
+    Error(JobError),
+}
+
+json_newtype!(JobResult);
+json_newtype!(Job);
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, Serialize, Deserialize)]
+#[sea_orm(table_name = "queue")]
+pub struct Model {
+    #[sea_orm(primary_key, auto_increment = false)]
+    pub id: Uuid,
+    #[serde(with = "time::serde::iso8601")]
+    pub created_at: OffsetDateTime,
+    #[serde(with = "time::serde::iso8601")]
+    pub updated_at: OffsetDateTime,
+    #[serde(default, with = "time::serde::iso8601::option")]
+    pub scheduled_at: Option<OffsetDateTime>,
+    pub failure_count: i32,
+    pub status: JobStatus,
+    pub job: Job,
+    pub result: Option<JobResult>,
```

Comment:
It seems unfortunate to have both `JobStatus` and `JobResult` fields here because now what if you have `JobResult = Complete` but `JobStatus = Pending`? However I understand that serializing to SQL means you sometimes have to put up with some ugliness in struct definitions so I can live with this.

Reply:
Yeah, JobStatus is effectively an optimization and possibly a premature one for the load we'll likely see. The JobStatus is currently intended as authoritative and the `JobResult` is informational and unused for any queue behavior.

Reply:
If we don't think there's a significant performance win to be had by having two columns, then my preference is a single field so that we can make illegal states impossible to represent and then we'll need fewer tests. Something like:

```
enum JobStatus {
    Pending,
    Success(Option<Uuid>), // Some(uuid) if a child job was spawned, None otherwise
    Failed(JobError),
}
```

Reply:
I think this may be a failure in documentation. It is perfectly valid to have a pending job with a job error result. That indicates that the product of the last execution was an error, but that we intend to retry